src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/DateFormatTest.java

```diff
@@ -88,15 +91,47 @@ public void orderByTest() {
     }
 
     @Test
-    @Ignore("06/27/2019: During implementing of ORDER BY date_format found that this fails as well. " +
-            "Will investigate after order by fix submitted, to scope amount of fixes.")
-    public void orderByWithGroupByTest() {
+    public void groupByWithDescOrder() throws SqlParseException {
         String query = "SELECT date_format(utc_time, 'dd-MM-YYYY'), count(*) " +
                 "FROM kibana_sample_data_logs " +
                 "GROUP BY date_format(utc_time, 'dd-MM-YYYY') " +
                 "ORDER BY date_format(utc_time, 'dd-MM-YYYY') DESC";
 
+        JSONObject sort = getAggregationOrder(query);
+        assertThat(sort.getString("_key"), is("desc"));
+    }
+
+    @Test
+    public void groupByWithAscOrder() throws SqlParseException {
+        String query = "SELECT date_format(utc_time, 'dd-MM-YYYY'), count(*) " +
+                "FROM kibana_sample_data_logs " +
+                "GROUP BY date_format(utc_time, 'dd-MM-YYYY') " +
+                "ORDER BY date_format(utc_time, 'dd-MM-YYYY')";
+
+        JSONObject sort = getAggregationOrder(query);
+        assertThat(sort.getString("_key"), is("asc"));
+    }
+
+    public JSONObject getAggregationOrder(String query) throws SqlParseException {
         Select select = getSelect(query);
+
+        Client client = mock(Client.class);
+        AggregationQueryAction queryAction = new AggregationQueryAction(client, select);
+
+        String elasticDsl = queryAction.explain().explain();
+        JSONObject elasticQuery = new JSONObject(elasticDsl);
+
+        JSONObject aggregations = elasticQuery.getJSONObject("aggregations");
+        String dateFormatAggregationKey = getScriptAggregationKey(aggregations, "date_format");
+
+        return aggregations.getJSONObject(dateFormatAggregationKey).getJSONObject("terms").getJSONObject("order");
+    }
+
+    public static String getScriptAggregationKey(JSONObject aggregation, String prefix) {
```

Comment:
nit: need a little more reformat in this method (don't know how to add multiple-line comment here...)